frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts

```diff
@@ -1,28 +1,67 @@
+import { Intent } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
+import { useEffect } from "react";
+import { QueryCache } from "react-query";
 import {
   CONVERSION_STATUS,
+  Dataset,
   DatasetUploadStatus,
   UPLOAD_STATUS,
   VALIDATION_STATUS,
 } from "src/common/entities";
+import { USE_DATASET_STATUS } from "src/common/queries/datasets";
+import DatasetUploadToast from "src/views/Collection/components/DatasetUploadToast";
+
+export enum FAILED_RETURN_TYPE {
+  UPLOAD = "UPLOAD",
+  VALIDATION = "VALIDATION",
+  CONVERSION = "CONVERSIOn",
+}
 
 export type FailReturn = {
   isFailed: boolean;
-  error?: VALIDATION_STATUS | UPLOAD_STATUS;
+  error?: VALIDATION_STATUS | UPLOAD_STATUS | CONVERSION_STATUS;
+  type?: FAILED_RETURN_TYPE;
 };
 
 export function checkIfFailed(datasetStatus: DatasetUploadStatus): FailReturn {
```

Comment:
enrich `checkIfFailed()` to include error type, so the tooltip can distinguish different types of `FAILED`